pages/_app.js

```diff
@@ -7,11 +7,33 @@ import { authenticatedFetch } from "@shopify/app-bridge-utils";
 import "@shopify/polaris/dist/styles.css";
 import translations from "@shopify/polaris/locales/en.json";
 
+function userLoggedInFetch(app) {
+  const fetchFunction = authenticatedFetch(app);
+
+  return async (uri, options) => {
+    const response = await fetchFunction(uri, options);
+
+    if (
+      response.headers.get("X-Shopify-API-Request-Failure-Reauthorize") === "1"
+    ) {
+      const authUrlHeader = response.headers.get(
+        "X-Shopify-API-Request-Failure-Reauthorize-Url"
+      );
+
+      const redirect = Redirect.create(app);
+      redirect.dispatch(Redirect.Action.APP, authUrlHeader || `/auth`);
+      return null;
```

Comment:
No issues here, just wondering why do we return null here?

Reply:
We may not need that, let me test it!

Reply:
Tested it, we actually do need to return `null` here, otherwise the redirect fails because the Apollo client will try to parse the response.